tests/test_model_signature.py

```diff
@@ -38,7 +38,11 @@ def test_generic_model_signature():
     class Model(BaseModel, Generic[T]):
         a: T
 
-    sig = signature(Model[int])
+    IntModel = Model[int]
+
+    assert IntModel.model_validate({'a': '1'}).a == 1
+
+    sig = signature(IntModel)
```

Comment:
It looks like the cause of this is that the following code is firing on this branch from `_model_construction.py`:

```
    if collect_invalid_schemas(schema):
        set_model_mocks(cls, cls_name)
        return False
```

I suggest we understand this further - smells like a possible bug.

Reply:
Let's xfail instead of modify, I will dig into this in parallel.